### PR TITLE
fix for issue #13837 External profile image URL wrongly redirect to default avatar URL for all-digits usernames

### LINKED ISSUE #13837: External profile image URL wrongly redirect to default avatar URL for all-digits usernames
I've just setup a new sever (using the Yunohost app).  When I follow someone on another server, the user only sees the default avatar and not my actual profile photo. Similarly, if someone searches for me from another server to follow me, they do not see my profile photo; all the other profile info is shown.

Is there anything I can check?  thanks

Friendica, version 2023.05.

Thanks

### PR DESCRIPTION
**Pull request by Raroun**
fixes #13837 
Added a check for non us-ascii characters and leading digits to the register module

### CODE REVIEW

File: src/Module/Register.php [hunk 1]
```diff
@@ -284,7 +284,21 @@ protected function post(array $request = [])
...
+		
+		//Check if nickname contains only US-ASCII and do not start with a digit
+		if (!preg_match('/^[a-zA-Z][a-zA-Z0-9]*$/', $arr['nickname'])) {
+        		if (is_numeric(substr($arr['nickname'], 0, 1))) {
+				Logger::info('Nickname with a leading digit not allowed', $arr);
+				DI::sysmsg()->addNotice(DI::l10n()->t("Nickname cannot start with a digit."));
+        		} else {
+				Logger::info('nickname with non us-ascii characters not allowed', $arr);
+				DI::sysmsg()->addNotice(DI::l10n()->t("Nickname can only contain US-ASCII characters."));
+			}
```
Comment (MrPetovan) [inline src/Module/Register.php]: Logger calls are unwarranted when we present the error message back to the user. If an admin stumbles on these calls, there's nothing then can do about it, it has already been taking care of by the System Messages feature.
```suggestion
        		if (is_numeric(substr($arr['nickname'], 0, 1))) {
				DI::sysmsg()->addNotice(DI::l10n()->t("Nickname cannot start with a digit."));
        		} else {
				DI::sysmsg()->addNotice(DI::l10n()->t("Nickname can only contain US-ASCII characters."));
			}
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]
Since you added a couple translation strings, please run `bin/run_xgettext.sh` from the Friendica root folder and commit the changes to the same branch.

**Comment by Raroun** [comment]
thanks for clarification with the user messages :)

**Review by MrPetovan — APPROVED** [review]

**Comment by MrPetovan** [comment]
Thank you for your work!

**Pull request merged**